File: 746-min-cost-climbing-stairs/solution.cpp
// Runtime: 3ms, Beats 80.13% of users with Python3
// Memory: 16.35MB, Beats 37.07% of users with Python3

class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int n=cost.size();
        vector<int> dp(n);
        for (int i=0;i<n;i++)
        {
            if (i<2) dp[i] = cost[i];
            else {
                dp[i] = cost[i] + min(dp[i-1],dp[i-2]);
            }
        }
        return min(dp[n-1],dp[n-2]);
    }
};
        
                

File: 746-min-cost-climbing-stairs/solution.py
# Runtime: 41ms, Beats 99.03% of users with Python3
# Memory: 16.68MB, Beats 81.98% of users with Python3

class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        prevprev = cost[0]
        prev = cost[1]
        for i in range (2,n):
            curr = cost[i]+min(prev,prevprev)
            prevprev = prev
            prev = curr
        return min(prev,prevprev)
        
                
